### PR TITLE
Support `last_value` window function

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### What problem does this PR solve?

Issue Number: ref #7376

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/WindowFunctions/IWindowFunction.cpp [hunk 1]
```diff
@@ -133,6 +133,116 @@ struct WindowFunctionRowNumber final : public IWindowFunction
...
+        size_t function_index,
+        const ColumnNumbers & arguments)
+    {
+        IColumn & to = *action.blockAt(action.current_row).output_columns[function_index];
+        const auto & value_column = *action.inputAt(action.current_row)[arguments[0]];
+        const auto & value_field = value_column[action.current_row.row];
+        to.insert(value_field);
+    }
+
+    void processFrameTypeUnbounded(
```
Comment (SeaRise) [inline dbms/src/WindowFunctions/IWindowFunction.cpp:195]: It seems that all cases can be handled by this function, no need to distinguish which frame type.

Comment (xzhangxian1008) [reply dbms/src/WindowFunctions/IWindowFunction.cpp:195]: > It seems that all cases can be handled by this function, no need to distinguish which frame type.

Sry, it's a bad name and the function will be renamed to `processBoundryTypeUnbounded`.

File: dbms/src/DataStreams/WindowBlockInputStream.cpp [hunk 2]
```diff
@@ -642,22 +642,41 @@ void WindowBlockInputStream::appendInfo(FmtBuffer & buffer) const
...
+    row_num.row = 0;
+    ++row_num.block;
+}
+
+RowNumber WindowTransformAction::getPreviousRowNumber(const RowNumber & row_num) const
+{
+    assert(row_num.block >= first_block_number);
+    assert(!(row_num.block == 0 && row_num.row == 0));
+
+    RowNumber prev_row_num = row_num;
```
Comment (SeaRise) [inline dbms/src/DataStreams/WindowBlockInputStream.cpp:668]: what happen for `[0, 0]`?

Comment (xzhangxian1008) [reply dbms/src/DataStreams/WindowBlockInputStream.cpp:668]: > what happen for `[0, 0]`?

Once we enter this function, row_num should be >[0, 0], or this must be a logic error.

**Review by SeaRise — APPROVED** [review]
lgtm

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**